utils/components-testkit/src/main/scala/pl/touk/nussknacker/engine/util/test/TestScenarioRunner.scala

```diff
@@ -50,6 +50,8 @@ trait TestScenarioRunnerBuilder[R <: TestScenarioRunner, B <: TestScenarioRunner
 
   def withExtraComponents(components: List[ComponentDefinition]): B
 
+  def withGlobalVariables(globalVariables: Map[String, WithCategories[AnyRef]]): B
```

Comment:
Please remember to extend https://nussknacker.io/documentation/docs/developers_guide/Testing/ See how it describes adding new components mechanism

Reply:
Good point, I event didn't know about this page :)